conan/tools/gnu/pkgconfigdeps/pkgconfigdeps.py

```diff
@@ -11,8 +11,13 @@ def __init__(self, conanfile):
     def content(self):
         """Get all the *.pc files content"""
         pc_files = {}
+        # Get all the dependencies
         host_req = self._conanfile.dependencies.host
-        for _, dep in host_req.items():
+        build_req = self._conanfile.dependencies.direct_build
+        test_req = self._conanfile.dependencies.test
+        for _, dep in list(host_req.items()) + list(build_req.items()) + list(test_req.items()):
```

Comment:
is there any risk of pkg-config files from different contexts overriding each other? like `build_requiries(protobuf)` + `requires(protobuf)`

Reply:
Yes, but that's a problem that you could even have with other components. Have a look at this PR https://github.com/conan-io/conan/pull/10263 So, you should change that `pkg_config_name`/`pkg_config_aliases` property.